packages/gatsby/src/redux/nodes.ts

```diff
@@ -180,7 +177,7 @@ export const ensureIndexByTypedChain = (
 
   if (nodeTypeNames.length === 1) {
     getNodesByType(nodeTypeNames[0]).forEach(node => {
-      addNodeToFilterCache(node, chain, filterCache, resolvedNodesCache)
+      addNodeToFilterCache(node, node, chain, filterCache, resolvedNodesCache)
```

Comment:
The node, node is a bit hard to wrap your head around without looking at the function.
```suggestion
      addNodeToFilterCache(node, node, chain, filterCache, resolvedNodesCache)
```

Reply:
Yeah :(

It's either this or duplication for `elemMatch`. For `elemMatch`, it will call this function once the `IDbQueryElemMatch` reaches the final `IDbQueryQuery` step. But that path doesn't start at the node root, but at the current step in resolving. At the same time we also need the original node to put it in the cache.

Do you have suggestions to improve this?

Reply:
I realized I could make it default to node if no root object was given. So that's what I'll do :)